enter.c

```diff
@@ -468,6 +527,45 @@ int enter(struct entry_settings *opts)
 		}
 	}
 
+	/* Only mount a a cgroup hierarchy over sys/fs/cgroup if:
+	 *  1) The user has not specified --no_cgroup_remount
+	 *  2) The mount namespaces are being unshared
+	 *  3) The cgroup namespaces are being unshared
+	 */
+	if (!opts->no_cgroup_remount && mnt_unshare && cgroup_unshare) {
+		int rootfd = open(root, O_PATH, 0);
+		if (rootfd == -1) {
+			err(1, "open %s", root);
+		}
+
+		struct stat cgroupst = { .st_dev = 0 };
+		if (fstatat(rootfd, "sys/fs/cgroup", &cgroupst, AT_NO_AUTOMOUNT | AT_SYMLINK_NOFOLLOW) == -1 && errno != ENOENT) {
+			err(1, "fstatat %s/sys/fs/cgroup", root);
+		}
+
+		struct stat rootst;
+		if (fstat(rootfd, &rootst) == -1) {
+			err(1, "fstat %s", root);
+		}
+
+		close(rootfd);
+
+		// Is /sys/fs/cgroup already a mountpoint?
+		if (cgroupst.st_dev != 0 && cgroupst.st_dev != rootst.st_dev) {
+			const char *target = makepath("%s/sys/fs/cgroup", root);
+
+			// Mount tmpfs over /sys/fs/cgroup
```

Comment:
Comments should explain the why, not the what, otherwise they end up redundant -- I can read the mount("tmpfs", target, "tmpfs", 0, NULL) and gather the exact same information as your comment.

What would be more interesting is explaining why mounting a tmpfs over the destination is necessary.

Reply:
Added clarifying comments based on our discussion.